src/plugins/dashboard/public/dashboard_top_nav/internal_dashboard_top_nav.tsx

```diff
@@ -195,7 +195,9 @@ export function InternalDashboardTopNav({
    */
   useEffect(() => {
     onAppLeave((actions) => {
-      if (
+      if (hasUnsavedChanges && !embeddableService.getStateTransfer().isTransferInProgress) {
+        return actions.skip();
+      } else if (
```

Comment:
I think the `skip` action is also overkill. Can you come up with a situation where we ever hit the `else if` condition with this current code?